src/ClientMessageHandler.js

```diff
@@ -0,0 +1,83 @@
+
+const logger = require('./logging');
+const storeFile = require('./store/file');
+
+const messageTypes = {
+    SequenceNumber: 'sn'
+};
+
+/**
+ * This handles sending the messages to the frontend
+ */
+class ClientSink {
+    /**
+     * @param tempPath {string}
+     * @param sequenceNumberSendingInterval {number}
+     */
+    constructor({ statssessionid, tempPath, sequenceNumberSendingInterval, demuxSink, client }) {
+        this.statssessionid = statssessionid;
+        this.tempPath = tempPath;
+        this.sequenceNumberSendingInterval = sequenceNumberSendingInterval;
+        this.demuxSink = demuxSink;
+        this.client = client;
+    }
+
+    /**
+     *  Sends the last sequence number from demuxSink or reads from the dump file
+    */
+    sendLastSequenceNumber() {
+        logger.info('Sending last sequence number from demuxSink');
+        let sequenceNumber = 0;
+
+        if (this.demuxSink.lastSequenceNumber > 0) {
+            sequenceNumber = this.demuxSink.lastSequenceNumber;
+        } else {
+            sequenceNumber = this._getLastSequenceNumberFromDump();
```

Comment:
If the dump file does not exist, will `_getLastSequenceNumberFromDump` throw an error? If not, when do we stop the loop?

Reply:
I added a condition to stop the loop when the connection is closed.